Reqnroll/TestRunnerManager.cs

```diff
@@ -203,32 +203,48 @@ private ITestRunner GetTestRunnerWithoutExceptionHandling()
         return testRunner;
     }
 
+    private async Task FireRemainingAfterFeatureHooks()
+    {
+        var testWorkerContainers = _availableTestWorkerContainers.Concat(_usedTestWorkerContainers).ToArray();
+        foreach (var testWorkerContainer in testWorkerContainers)
+        {
+            var contextManager = testWorkerContainer.Key.Resolve<IContextManager>();
+            if (contextManager.FeatureContext != null)
+            {
+                var testRunner = testWorkerContainer.Key.Resolve<ITestRunner>();
+                await testRunner.OnFeatureEndAsync();
+            }
+        }
+    }
+
     public virtual async Task DisposeAsync()
     {
         if (Interlocked.CompareExchange(ref _wasDisposed, 1, 0) == 0)
         {
+            await FireRemainingAfterFeatureHooks();
```

Comment:
Could this result in a late release of the feature context?

For example
- Feature A runs for 1 second
- Feature B runs 60 seconds

Feature A's `AfterFeature' hook would only run after 59 seconds or?

I don't think this happens often or is a really bad thing (in MsTest this was always the case before #128). Just wanted to point out the change.

Reply:
@obligaron Yes, this is correct. Such delayed release of feature context could happen. (But it always runs before the After Test Run hook even in that case.)